### DISCUSSION TITLE
How to run the macdrive apps in background? without an active terminal session?

### DISCUSSION THREAD

**Discussion by tomvictor** [progrium/darwinkit#142] [General]
Hi @progrium 
Thanks for the awesome contribution.
Just wanted to know whether there is any work arounds to run the app in background. The application is running perfectly fine with an active terminal session now. Currently I am invoking the app by cmd below. 

```bash
nohup app &
```

Thanks
Regards,
Tom Victor

**Comment by progrium** [comment]
Hi! 

Once you package it into an app bundle, it can launch without a terminal. We'll mention how in the docs at some point or even build a tool to help with this. For now: https://stackoverflow.com/questions/1596945/building-osx-app-bundle

That help?

**Reply by tomvictor** [reply]
Hi @progrium!

yes you are right, packaging is the best way then I found something interesting. And finally I found a simple packaging solution using fyne (https://developer.fyne.io/). Everything is working now, now I will add more features and ship for public use. The app and source code is available on https://github.com/tomvictor/linu 

Thanks

Regards,
@tomvictor

**Reply by progrium** [reply]
Is that using their CLI? What was the command to do it?

**Reply by tomvictor** [reply]
hi @progrium 

Installation

```bash
go install fyne.io/fyne/v2/cmd/fyne@latest
```

Packaging

```bash
fyne package -os darwin -icon icon.png
```

This will package the apps but it is not fully qualified to ship. When I tried to install in a different Mac it failed. 
I found an intrsting doc in the arduino-IDE build script. https://github.com/arduino/arduino-ide/blob/f6a43254f5c416a2e4fa888875358336b42dd4d5/.github/workflows/build.yml#L79

fyne doc: https://developer.fyne.io/started/packaging

**Comment by programmingkidx** [comment]
I really don't like the fact you need to install another GUI framework to make your application. Now I am wondering if we should make our own tooling or use another tool. Whatever solution we come up with I will be happy to add the documentation for it in the Wiki.

**Reply by progrium** [reply]
I have some code to do this, but once you start talking about bundling you usually end up talking about signing and entitlements which are tricky. But it's on the roadmap in my head.

**Reply by programmingkidx** [reply]
This program probably would be pretty easy to make. It might run like this:
`create_bundle -name <name of program> -binary <binary> -setIcon <picture>`

**Reply by progrium** [reply]
Something like that yes. It should assemble common Plist configuration, but also allow you to provide any Plist option. There are other things it can do to help. My apptron project has an undocumented first pass at this, which basically wraps some code that somebody else put together. But this would be a good place to start: https://github.com/tractordev/apptron/blob/main/cmd/apptron/bundle/bundler/bundler.go

This code is a hacky conversion of a program into a package. It could be refined into a proper package and then exposed as a command line tool separately. I recommend also reading the first link in the source. And this is the gist it was based on: https://gist.github.com/mholt/11008646c95d787c30806d3f24b2c844
